common/src/main/java/com/lowdragmc/lowdraglib/gui/widget/SlotWidget.java

```diff
@@ -265,28 +268,53 @@ public void drawInBackground(@Nonnull GuiGraphics graphics, int mouseX, int mous
     public boolean mouseClicked(double mouseX, double mouseY, int button) {
         if (slotReference != null && isMouseOverElement(mouseX, mouseY) && gui != null) {
             var stack = slotReference.getItem();
-            if (!(canPutItems && stack.isEmpty() || canTakeItems && !stack.isEmpty())) return false;
-            ModularUIGuiContainer modularUIGui = gui.getModularUIGui();
-            boolean last = modularUIGui.getQuickCrafting();
-            InputConstants.Key mouseKey = InputConstants.Type.MOUSE.getOrCreate(button);
-            HOVER_SLOT = slotReference;
-            gui.getModularUIGui().superMouseClicked(mouseX, mouseY, button);
-            HOVER_SLOT = null;
-            if (last != modularUIGui.getQuickCrafting()) {
-                modularUIGui.dragSplittingButton = button;
-                if (button == 0) {
-                    modularUIGui.dragSplittingLimit = 0;
-                } else if (button == 1) {
-                    modularUIGui.dragSplittingLimit = 1;
-                } else if (Minecraft.getInstance().options.keyPickItem.matchesMouse(mouseKey.getValue())) {
-                    modularUIGui.dragSplittingLimit = 2;
+            if (canPutItems && stack.isEmpty() || canTakeItems && !stack.isEmpty()) {
+                ModularUIGuiContainer modularUIGui = gui.getModularUIGui();
+                boolean last = modularUIGui.getQuickCrafting();
+                InputConstants.Key mouseKey = InputConstants.Type.MOUSE.getOrCreate(button);
+                HOVER_SLOT = slotReference;
+                gui.getModularUIGui().superMouseClicked(mouseX, mouseY, button);
+                HOVER_SLOT = null;
+                if (last != modularUIGui.getQuickCrafting()) {
+                    modularUIGui.dragSplittingButton = button;
+                    if (button == 0) {
+                        modularUIGui.dragSplittingLimit = 0;
+                    } else if (button == 1) {
+                        modularUIGui.dragSplittingLimit = 1;
+                    } else if (Minecraft.getInstance().options.keyPickItem.matchesMouse(mouseKey.getValue())) {
+                        modularUIGui.dragSplittingLimit = 2;
+                    }
+                }
+                return true;
+            } else if (LDLib.isEmiLoaded()) {
+                if (getXEICurrentIngredient() instanceof EmiStack emiStack) {
```

Comment:
May be its necessary to implement `EMIStack` in the `getXEICurrentIngredient` ?

Reply:
its was implemented in SlotWidget but was missing the TankWidget one